sycl/test/e2e_test_requirements/no-unsupported-without-info.cpp

```diff
@@ -0,0 +1,547 @@
+// This test is intended to ensure that we have no tests marked as
+// UNSUPPORTED without an information added to a test.
+// For more info see: sycl/test-e2e/README.md
+//
+// The format we check is:
+// UNSUPPORTED: lit,features
+// UNSUPPORTED-TRACKER: [GitHub issue URL|Internal tracker ID]
+// *OR*
+// UNSUPPORTED: lit,features
+// UNSUPPORTED-INTENDED: explanation why the test isn't intended to be run with this feature
+//
+// GitHub issue URL format:
+//     https://github.com/owner/repo/issues/12345
+//
+// Internal tracker ID format:
+//     PROJECT-123456
+//
+// REQUIRES: linux
+//
+// Explanation of the command:
+// - search for all "UNSUPPORTED" occurrences, display line with match and the next one
+//   -I, --include to drop binary files and other unrelated files
+// - in the result, search for "UNSUPPORTED" again, but invert the result - this
+//   allows us to get the line *after* UNSUPPORTED
+// - in those lines, check that UNSUPPORTED-TRACKER or UNSUPPORTED-INTENDED is
+//   present and correct. Once again, invert the search to get all "bad" lines
+//   and save the test names in the temp file
+// - make a final count of how many ill-formatted directives there are and
+//   verify that against the reference
+// - ...and check if the list of improperly UNSUPPORTED tests needs to be updated.
+//
+// RUN: grep -rI "UNSUPPORTED:" %S/../../test-e2e \
+// RUN: -A 1 --include=*.c --include=*.cpp --no-group-separator | \
+// RUN: grep -v "UNSUPPORTED:" | \
+// RUN: grep -Pv "UNSUPPORTED-TRACKER:\s+(?:https://github.com/[\w\d-]+/[\w\d-]+/issues/[\d]+)|(?:[\w]+-[\d]+)|(?:UNSUPPORTED-INTENDED:\s*.+)" > %t
```

Comment:
```suggestion
// RUN: grep -Pv "UNSUPPORTED-TRACKER:\s+(?:https://github.com/[\w\d-]+/[\w\d-]+/issues/[\d]+)|(?:[\w]+-[\d]+)|(?:UNSUPPORTED-INTENDED:\s\w.*)" > %t
```

- to require nice formatting (i.e. exactly one spacing symbol after `-INDENDED:`
- to require at least single alphanumeric symbol as an explanation. Not that it would prevent all possible malformed explanations, but that's at least something.

But that is mostly a minor comment, I believe that formatting and reasoning will be up to reviewers anyways.